variation-commons-core/src/main/java/uk/ac/ebi/eva/commons/core/models/ws/VariantWithSamplesAndAnnotation.java

```diff
@@ -51,6 +54,22 @@ public VariantWithSamplesAndAnnotation(String chromosome, long start, long end,
         sourceEntries = new HashMap<>();
     }
 
+    public VariantWithSamplesAndAnnotation(Variant variant) {
+        super(variant.getChromosome(), variant.getStart(), variant.getEnd(), variant.getReference(), variant.getAlternate());
+
+        Map<String, VariantSourceEntryWithSampleNames> sourceEntries = new HashMap<>();
+        for (VariantSourceEntry entry : variant.getSourceEntries()) {
```

Comment:
This block could be extracted to a copy constructor, using the same design as for objects in the `IVariant` hierarchy.